### PR TITLE
bytecode: Decode opcodes with immediate arguments

### PR DESCRIPTION
**Pull request by rodiazet**
- Support decoding all opcodes with imm arguments in `bytecode` library

### CODE REVIEW

File: test/utils/bytecode.hpp [hunk 1]
```diff
@@ -550,16 +550,17 @@ inline std::string decode(bytes_view bytecode)
         {
             s += std::string{" + OP_"} + name;
 
-            if (opcode >= OP_PUSH1 && opcode <= OP_PUSH32)
+            if (size_t imm_data_size = evmone::instr::traits[opcode].immediate_size;
```
Comment (chfast) [inline test/utils/bytecode.hpp:553]:
```suggestion
            if (const size_t imm_data_size = evmone::instr::traits[opcode].immediate_size;
```

File: test/utils/bytecode.hpp [hunk 2]
```diff
@@ -550,16 +550,17 @@ inline std::string decode(bytes_view bytecode)
...
+                imm_data_size > 0)
             {
-                const auto push_data_start = it + 1;
-                const auto push_data_size =
-                    std::min(static_cast<std::size_t>(opcode - OP_PUSH1 + 1),
-                        static_cast<std::size_t>(bytecode.end() - push_data_start));
-                if (push_data_size != 0)
+                const auto imm_data_start = it + 1;
+                imm_data_size = std::min(static_cast<std::size_t>(imm_data_size),
```
Comment (chfast) [inline test/utils/bytecode.hpp:557]:
```suggestion
                imm_data_size = std::min(imm_data_size,
```

File: test/utils/bytecode.hpp [hunk 3]
```diff
@@ -550,16 +550,17 @@ inline std::string decode(bytes_view bytecode)
...
+                    static_cast<std::size_t>(bytecode.end() - imm_data_start));
```
Comment (chfast) [inline test/utils/bytecode.hpp:558]:
```suggestion
                    static_cast<size_t>(bytecode.end() - imm_data_start));
```

File: test/utils/bytecode.hpp (hunk 1 above)
Comment (chfast) [reply test/utils/bytecode.hpp:553]: I also think we could name it just `imm_size` unless it is already commonly named `imm_data` in other places.

**Review by chfast — CHANGES_REQUESTED** [review]
Also add a test for e.g. `RJUMP`.

File: test/utils/bytecode.hpp (hunk 1 above)
Comment (rodiazet) [reply test/utils/bytecode.hpp:553]: OK for naming. `imm_size` cannot be `const`. It's changed below.

**Review by chfast — APPROVED** [review]

**Pull request merged**